clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/agent/KubernetesCacheDataConverter.java

```diff
@@ -166,6 +159,9 @@ private static void addLogicalRelationships(
       Moniker moniker,
       boolean hasClusterRelationship) {
     String application = moniker.getApp();
+    if (Strings.isNullOrEmpty(application)) {
+      return;
+    }
```

Comment:
In what case would this happen? Seems like something worth logging _at least_ a WARN about?

Reply:
Hmm, I moved that condition down from where it was (outside the function itself). I'm not super familiar with the Frigga naming, we use the moniker annotations ourselves. Is it possible that resources could have names that don't result in them having an app name?

If so, a WARN would probably be too verbose, if we're reading all resources in a namespace. What do you think?

Reply:
Ah, I missed that this was just pulled down from the method above. In a Frigga world, it's impossible to have a resource that is not associated with an application, which is why I asked. In a Moniker world, I don't believe it makes any guarantees about the presence of any data from a resource name, so it's probably fine as you have it.